Gems/ROS2/Code/Source/Frame/ROS2Transform.cpp

```diff
@@ -38,7 +53,20 @@ namespace ROS2
         { // Only publish static transforms once
             return;
         }
-        ROS2Interface::Get()->BroadcastTransform(CreateTransformMessage(transform), m_isDynamic);
+        auto transformMessage = CreateTransformMessage(transform);
+        ROS2Interface::Get()->BroadcastTransform(transformMessage, m_isDynamic);
+
+        if (m_transformPublisher)
+        {
+            AZStd::string parent_id(transformMessage.header.frame_id.c_str());
+            AZStd::string child_frame_id(transformMessage.child_frame_id.c_str());
+            ROS2Names::StripNamespace(parent_id);
```

Comment:
This behavior of stripping namespace should be optional because this prevents simple tf merging / filtering.